StripePayments/StripePayments/Source/PaymentHandler/STPPaymentHandler.swift

```diff
@@ -1183,250 +1133,188 @@ public class STPPaymentHandler: NSObject {
-                                                                )
-                                                            }
-                                                        } else {
-                                                            transaction.doChallenge(
-                                                                with: presentingViewController,
-                                                                challengeParameters:
-                                                                    challengeParameters,
-                                                                challengeStatusReceiver: self,
-                                                                timeout: TimeInterval(
-                                                                    currentAction
-                                                                        .threeDSCustomizationSettings
-                                                                        .authenticationTimeout
-                                                                        * 60
-                                                                )
-                                                            )
-                                                        }
-
-                                                    },
-                                                    catch: { exception in
-                                                        self.currentAction?.complete(
-                                                            with: STPPaymentHandlerActionStatus
-                                                                .failed,
-                                                            error: self._error(
-                                                                for: .stripe3DS2ErrorCode,
-                                                                loggingSafeUserInfo: [
-                                                                    "exception": exception.description,
-                                                                ]
-                                                            )
-                                                        )
-                                                    },
-                                                    finallyBlock: {
-                                                    }
+                                    guard self._canPresent(
+                                        with: currentAction.authenticationContext,
+                                        error: &presentationError
+                                    ) else {
+                                        currentAction.complete(
+                                            with: .failed,
+                                            error: presentationError
+                                        )
```

Comment:
Sorry, hard to follow the code, but when we complete an action with `.failed`, it seems like this eventually gets caught on line 407, w/ an assertionFailure?  I'm guessing we probably want to call stpAssert on this line?